### PR TITLE
CXP-1283: New bucket for catalogs mapping

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/tests/Integration/Infrastructure/Storage/CatalogsMappingStorageTest.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public function testItChecksIfAFileExists(): void
+    {
+        $this->filesystem->write('db1079b6-f397-4a6a-bae4-8658e64ad47c_product.json', self::SCHEMA);
+
+        $exists = $this->storage->exists('db1079b6-f397-4a6a-bae4-8658e64ad47c_product.json');
+
+        $this->assertTrue($exists);
+    }
+
+    public function testItReturnAFileContents(): void
```
Comment (willy-ahva) [inline components/catalogs/back/tests/Integration/Infrastructure/Storage/CatalogsMappingStorageTest.php:64]:
```suggestion
    public function testItReturnsAFileContents(): void
```

**Review by willy-ahva — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Storage/CatalogsMappingStorageTest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        parent::setUp();
+
+        $this->purgeData();
+
+        $this->storage = self::getContainer()->get(CatalogsMappingStorage::class);
+        $this->filesystem = self::getContainer()->get('oneup_flysystem.catalogs_mapping_filesystem');
+    }
+
+    public function testItCreatesAFile(): void
```
[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/tests/Integration/Infrastructure/Storage/CatalogsMappingStorageTest.php:46]: I understand the confusion, but `$this->storage` is the class being tested.
So the test goes like this:
1) Write a file using CatalogsMappingStorage
2) Read the file content using Flysystem directly, bypassing the interface
3) Assert that the file has the expected content.

File: config/packages/prod/oneup_flysystem.yml [hunk 3]
```diff
@@ -12,3 +12,6 @@ oneup_flysystem:
         local_storage_adapter:
             local:
                 location: '/'
+        catalogs_mapping_adapter:
```
Comment (masacc) [inline config/packages/prod/oneup_flysystem.yml:15]: just think about it: what about naming it `app_catalogs_mapping` to clearly identify the type of catalog?

**Review by masacc — APPROVED** [review]

File: config/packages/prod/oneup_flysystem.yml (hunk 3 above)
Comment (tseho) [reply config/packages/prod/oneup_flysystem.yml:15]: In theory, Catalogs is not reserved to Apps, it's almost a standalone feature.

**Pull request merged**